components/builder-depot/src/backend/s3.rs

```diff
@@ -169,45 +193,53 @@ impl S3Handler {
         }
     }
 
-    fn single_upload(&self, key: String, hart: File, path_attr: &str) -> Result<()> {
+    fn single_upload(&self, key: &str, hart: File, path_attr: &PathBuf) -> Result<()> {
```

Comment:
Consider changing path_attr type to AsRef<PathBuf> so anything convertible to a Pathbuf will just work..